xmodule/js/src/video/037_video_transcript_feedback.js

```diff
@@ -0,0 +1,254 @@
+                    if (this.currentFeedback === false) {
+                        this.sendFeedbackForCurrentTranscript(null);
+                    } else {
+                        this.sendFeedbackForCurrentTranscript(false);
+                    }
+                },
+
+                sendFeedbackForCurrentTranscript: function(feedbackValue) {
+                    var self = this;
+                    var url = this.aiTranslationsUrl + '/transcript-feedback/';
+                    $.ajax({
+                        url: url,
+                        type: 'POST',
+                        dataType: 'json',
+                        data: {
+                            transcript_language: this.currentTranscriptLanguage,
+                            video_uuid: this.videoId,
+                            user_id: this.userId,
+                            value: feedbackValue,
+                        },
+                        success: function(data) {
+                            if (data && data.value === true) {
+                                self.markAsPositiveFeedback();
+                                self.currentFeedback = true;
+                            } else {
+                                if (data && data.value === false) {
+                                    self.markAsNegativeFeedback();
+                                    self.currentFeedback = false;
+                                } else {
+                                    self.markAsEmptyFeedback();
+                                    self.currentFeedback = null;
+                                }
+                            }
+                        },
+                        error: function() {
+                            self.markAsEmptyFeedback();
+                            self.currentFeedback = null;
+                        }
+                    });
+                },
+
+                onHideLanguageMenu: function() {
+                    var newLanguageSelected = this.getCurrentLanguage();
+                    if (this.currentTranscriptLanguage !== newLanguageSelected) {
+                        this.currentTranscriptLanguage = this.getCurrentLanguage();
+                        this.instantiateWidget();
+                    }
+                },
+
+                getCurrentLanguage: function() {
+                    var language = this.state.lang;
+                    return language;
+                },
+
+                instantiateWidget: function() {
+                    var self = this;
+                    $.when(this.shouldShowWidget()).done(function(shouldShowWidget) {
+                        if (shouldShowWidget) {
+                            self.showWidget();
+                            self.getFeedbackForCurrentTranscript();
+                        } else {
+                            self.hideWidget();
+                        }
+                    })
+                },
+
+                shouldShowWidget: function() {
```

Comment:
I'm not sure I understand this function. It is named `shouldShowWidget` but makes an AJAX call, and that should somehow determine whether the widget is shown, but I cannot see from this what the criteria are. Is this supposed to also hide the widget when the user has already given feedback? And as far as checking whether the transcript is AI generated, as you see from my other comment this doesn't work correctly for me, probably because I have the URL here not set up correctly (I don't have the service locally, set the URL to stage, and get CORS errors). If the request fails as it does for me, however, should this really evaluate to true?

Reply:
The widget has to go through two conditions:
* One in the Django template level which evaluates the waffle flag.
* The other one at the JS file level which checks if the transcript selected for the current video was AI generated or not.

I'll check why it is not working as expected since the widget shouldn't be shown if the ai-translations service is not up.

Reply:
This should be fixed now

Reply:
Nice work. The widget is not shown any longer for non-ai transcripts.

The method name is still misleading: `shouldShowWidget` tells me to expect a boolean return.
Can you rename this to something like `loadAndSetVisibility` that encompasses what the function does?